server/platform/services/sharedchannel/service.go

```diff
@@ -277,3 +283,51 @@ func (scs *Service) notifyClientsForSharedChannelUpdate(channel *model.Channel)
 	messageWs.Add("channel_id", channel.Id)
 	scs.app.Publish(messageWs)
 }
+
+// isGlobalUserSyncEnabled checks if the global user sync feature is enabled
+func (scs *Service) isGlobalUserSyncEnabled() bool {
+	cfg := scs.server.Config()
+
+	if !cfg.FeatureFlags.EnableSyncAllUsersForRemoteCluster {
+		return false
+	}
+
+	// Skip if config explicitly disables the feature
+	if cfg.ConnectedWorkspacesSettings.SyncUsersOnConnectionOpen != nil &&
+		!*cfg.ConnectedWorkspacesSettings.SyncUsersOnConnectionOpen {
+		return false
+	}
+
+	return true
+}
+
+// tryScheduleGlobalUserSync schedules a task to sync all users with a remote cluster
+func (scs *Service) tryScheduleGlobalUserSync(rc *model.RemoteCluster) {
+	if !scs.isGlobalUserSyncEnabled() {
+		return
+	}
+
+	// Schedule the sync task
+	go func() {
+		// Create a special sync task with empty channelID
+		// This empty channelID is a deliberate marker for a global user sync task
```

Comment:
Maybe not for this PR, but we're going to need to document somewhere what shape each task has to avoid confusion, or even create a task type and be explicit about it